### ISSUE TITLE
Android Builds failing in RN 0.59

### ISSUE THREAD

**Issue by mrarlloyd** [segmentio/analytics-react-native#46]
Since upgrading to the recent React Native 0.59, my builds are failing because of this package:

```
FAILURE: Build failed with an exception.

* What went wrong:
The Android Gradle plugin supports only Kotlin Gradle plugin version 1.3.0 and higher.
The following dependencies do not satisfy the required version:
project ':@segment_analytics-react-native' -> org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.60```

**Comment by dmhood** [comment]
there's a pr for this (https://github.com/segmentio/analytics-react-native/pull/44) that hopefully a maintainer can look at soon

**Comment by kbrandwijk** [comment]
This is blocking us after the RN 0.59 upgrade, so I hope this can be done soon.

**Comment by jannainm** [comment]
If you're feeling dangerous here's an example of a shell script that can temporarily make the changes needed in the PR for android's build.gradle to get it working in RN 0.59.

Info/Disclaimers:
* This script is dangerous because it uses the sed command to make changes in place
* This works on OSX but for others you will want to replace `sed -i '' 's/pattern/replace/' filename` with `sed -i 's/pattern/replace/' filename`
* This script is probably poorly written but it works as an example for anyone as impatient as I am to get this PR in
* Works well as a `postinstall` command in your package.json so that after you yarn/npm install the changes are made immediately (that's how my CD system is able to build and deploy our app currently)
* Script is located in a `scripts/` directory within the root my app - so you would have package.json one level above this
* This has been tested but I make no claims to know that it's working perfectly - I am, however, getting events sent to segment and able to build on android which is enough  for my case
*  Refactor to apply it to your case, use it to learn, or laugh at how poorly it is written :)

`
#!/usr/bin/env bash

cd $(dirname "$0")

# If not called from scripts, then being sourced from appcenter in ../ (no need to move up if so)
INPUT_DIR=$(dirname "$0")
CALL_FOLDER=$(echo $INPUT_DIR | rev | cut -d "/" -f1 | rev)
if [ $CALL_FOLDER = "scripts" ]; then
  # echo "CDing up one folder to project root..."
  cd ../
fi
PROJECT_HOME=$(pwd)

PACKAGE_JSON_PATH=$PROJECT_HOME/package.json
SEGMENT_GRADLE=$PROJECT_HOME/node_modules/@segment/analytics-react-native/android/build.gradle

SEGMENT_VERSION=$(cat $PACKAGE_JSON_PATH | grep "\"@segment/analytics-react-native\"" | awk -F "beta." '{print $2}')
SEGMENT_VERSION="${SEGMENT_VERSION:0:1}"
if [ "$SEGMENT_VERSION" = "3" ]; then
  if [ -e $SEGMENT_GRADLE ]; then
    sed -i "" "s/classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.60'/classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.21'/" $SEGMENT_GRADLE
    sed -i "" "s/api 'com.segment.analytics.android:analytics:4.+'/implementation 'com.segment.analytics.android:analytics:4.+'/" $SEGMENT_GRADLE
    sed -i "" "s/api 'com.facebook.react:react-native:+'/implementation 'com.facebook.react:react-native:+'/" $SEGMENT_GRADLE
    sed -i "" "s/api 'org.jetbrains.kotlin:kotlin-stdlib:1.2.+'/implementation 'org.jetbrains.kotlin:kotlin-stdlib:1.3.+'/" $SEGMENT_GRADLE
  fi
fi
`

**Comment by kbrandwijk** [comment]
I'm using `patch-package`right now for this, which is a way safer and more controllable alternative than a 'blanket' script:

For `@segment/analytics-react-native`:
```diff
diff --git a/node_modules/@segment/analytics-react-native/android/build.gradle b/node_modules/@segment/analytics-react-native/android/build.gradle
index a0741fa..6834476 100644
--- a/node_modules/@segment/analytics-react-native/android/build.gradle
+++ b/node_modules/@segment/analytics-react-native/android/build.gradle
@@ -8,7 +8,7 @@ buildscript {
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.4'
-        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.60'
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.21'
     }
 }
 
@@ -41,5 +41,5 @@ dependencies {
     api 'com.segment.analytics.android:analytics:4.+'
 
     api 'com.facebook.react:react-native:+'
-    api 'org.jetbrains.kotlin:kotlin-stdlib:1.2.+'
+    api 'org.jetbrains.kotlin:kotlin-stdlib:1.3.+'
 }
```

For `@segment/analytics-react-native-firebase`:
```diff
diff --git a/node_modules/@segment/analytics-react-native-firebase/android/build.gradle b/node_modules/@segment/analytics-react-native-firebase/android/build.gradle
index 987d9f6..5cbc37a 100644
--- a/node_modules/@segment/analytics-react-native-firebase/android/build.gradle
+++ b/node_modules/@segment/analytics-react-native-firebase/android/build.gradle
@@ -8,7 +8,7 @@ buildscript {
     }
     dependencies {
         classpath 'com.android.tools.build:gradle:3.1.4'
-        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.2.60'
+        classpath 'org.jetbrains.kotlin:kotlin-gradle-plugin:1.3.21'
     }
 }
```

**Comment by kbrandwijk** [comment]
For the record: I do find it a bit ridiculous that we:
1. Need to use a _beta_ version of a package for production purposes
2. That we are now a couple of weeks into the latest RN version, and that there hasn't even been a single comment on this.

**Comment by ericketts** [comment]
@kbrandwijk hear hear, you expect to run in to some pain points with react-native, because its free and open source, and there are certain costs associated with achieving cross platform support -- but the majority of segment's RN lib a minimal façade on top of the native libs for the 2 major mobile platforms, its bemusing how much has managed to go awry in such a small amount of code.

**Comment by fathyb** [comment]
Hey everyone, we've released a bug fix in `0.0.1-beta.4`: https://github.com/segmentio/analytics-react-native/blob/master/CHANGELOG.md#001-beta4-2019-03-29

Please check it out and let us know if you're still experiencing issues.

**Comment by boxgames1** [comment]
@fathyb Could you release that version to npm? Thanks
